### DISCUSSION TITLE
Allow 'and' type declaration outside of same source file

### DISCUSSION THREAD

**Discussion by Xyncgas** [fsharp/fslang-suggestions#1274] [Type System]
If I want to write ``` type A and B``` I currently have to put them in the same source file with nothing in between
Ideally I can have multiple source file, and I can create type B before type A and still have the knowledge of type A through the keyword ```and```

It saves me the hassle to put them in the same source file because it's gonna get cramped if you expect people to put everything in a single file

**Comment by bartelink** [comment]
That's an unlikely thing to be added...
- how many files down would it need to look? compilation would be less efficient and things get ambiguous
- it would likely involves lots of impl work - the searches for symbols are constrained to within a file atm even in namespace/module `rec`
- C# already lets you do all this

Instead, I'd suggest:
- look if `namespace rec` and `module rec` work for you
- see if you can work with it the way its intended - IME it's never immediately obvious how to restructure things, but it's always possible and you tend to end up with a cleaner solution in the end.
